src/isolate.rs

```diff
@@ -69,6 +70,40 @@ impl IsolateState {
   }
 }
 
+pub struct Metrics {
+  pub ops_executed: i32,
+  pub bytes_recv: i32,
+  pub bytes_sent: i32,
+}
+
+impl Metrics {
+  pub fn new() -> Metrics {
+    Metrics {
+      ops_executed: 0,
+      bytes_recv: 0,
+      bytes_sent: 0
+    }
+  }
+
+  #[allow(dead_code)]
+  fn increment_ops_executed(mut self) {
```

Comment:
You likely want `&mut self` here and below. Plain `mut self` moves `self` into the function (`self` is "consumed"), while `&mut self` takes `self` by reference.

Reply:
Thanks, however when I do that I get 
```
error[E0596]: cannot borrow field of immutable binding as mutable
   --> ../../src/isolate.rs:307:5
    |
307 |     state.metrics.increment_ops_executed();
    |     ^^^^^^^^^^^^^ cannot mutably borrow field of immutable binding
```

Not sure how can I make `metrics` a mutable binding of `state` since it's field in struct

Reply:
I believe this will be fixed if you wrap the `metrics` field in a `Mutex` as suggested above.

The problem is that you have an `Arc` which has shared ownership, while mutating requires unique ownership. If you *know* that you are the only one holding the `Arc`, you can use [`Arc::get_mut`](https://doc.rust-lang.org/std/sync/struct.Arc.html#method.get_mut), but normally you don't know that so you need a `Mutex`.

Reply:
Thanks @bluetech! With your advice I managed to get it working!